### PR TITLE
Feature - Disable Frontend...

### PR DESCRIPTION
**Pull request by Raistlfiren**
Added the ability to disable the frontend through a configuration value and just show a blank page.

### CODE REVIEW

File: src/JSONAPIExtension.php [hunk 1]
```diff
@@ -63,11 +65,41 @@ public static function getSubscribedEvents()
...
+                if (strpos($routeName, 'jsonapi') === false) {
+                    $event->stopPropagation();
+                    $event->setController(
+                        function() {
+                            return new Response('', 200);
+                        }
+                    );
+                }
+            }
+        }
```
Comment (GwendolenLynch) [inline src/JSONAPIExtension.php]: Can you un-nest this. Early returns are far better for human reading of logical intent, and there is rarely a time where an if statement inside another is valid.

File: src/JSONAPIExtension.php [hunk 2]
```diff
@@ -63,11 +65,41 @@ public static function getSubscribedEvents()
             KernelEvents::EXCEPTION => [
                 ['error', 515],
             ],
+            KernelEvents::CONTROLLER => [
+                ['disableFrontend', -1025]
```
Comment (GwendolenLynch) [inline src/JSONAPIExtension.php]: This is massively high (late in terms of priority, but not the point :stuck_out_tongue: ). Does it really need to be?

File: src/JSONAPIExtension.php [hunk 3]
```diff
@@ -63,11 +65,41 @@ public static function getSubscribedEvents()
...
+            ]
         ];
 
         return $parentEvents + $localEvents;
     }
 
+    public function disableFrontend(FilterControllerEvent $event)
+    {
+        $request = $event->getRequest();
+        //$response = $event->getResponse();
```
Comment (GwendolenLynch) [inline src/JSONAPIExtension.php]: Remove commented code

File: src/JSONAPIExtension.php [hunk 4]
```diff
@@ -63,11 +65,41 @@ public static function getSubscribedEvents()
...
+
+        $container = $this->getContainer();
+
+        //Get route name
```
Comment (GwendolenLynch) [inline src/JSONAPIExtension.php]: Comment serves no purpose :wink:

File: src/JSONAPIExtension.php [hunk 5]
```diff
@@ -63,11 +65,41 @@ public static function getSubscribedEvents()
...
+        $routeName = $request->get('_route');
+
+        //Check if request is to frontend
+        if (Zone::isFrontend($request)) {
+            //Check if we should disable frontend based upon the configuration
+            if ($container['jsonapi.config']->isDisableFrontend()) {
+                //Only disable frontend routes, don't disable json routes
+                if (strpos($routeName, 'jsonapi') === false) {
+                    $event->stopPropagation();
```
Comment (GwendolenLynch) [inline src/JSONAPIExtension.php]: Not needed. You're setting a non-null value below

File: src/JSONAPIExtension.php [hunk 6]
```diff
@@ -63,11 +65,41 @@ public static function getSubscribedEvents()
...
+                    $event->setController(
+                        function() {
+                            return new Response('', 200);
+                        }
```
Comment (GwendolenLynch) [inline src/JSONAPIExtension.php]: You're better off throwing a `HttpException` here with a 403 or similar. This is just a nightmare to debug if for some reason it is being unexpectedly called … the Samuel L. Jackson Law of Assumption in full flight.

**Review by GwendolenLynch — CHANGES_REQUESTED** [review]

**Comment by Raistlfiren** [comment]
@GawainLynch  - You have reminded me once again why I submit PRs and have you review them. Do you have any job openings? I would love for you to be my boss. ;) :)

Seriously though, thank you @GawainLynch for the review. I will revise my code as requested, and I hope to push some changes within the next few days.

**Comment by GwendolenLynch** [comment]
No worries mate … Being you, I thought I'd go a bit harder than normal. :smiley_cat: 

Also, @xiaohutai is on leave, so someone has to harass people for no good reason :innocent:

**Comment by xiaohutai** [comment]
Hi @Raistlfiren ,

~I'm wondering why would you want this?~ This feature disables the website, so that you only get the JSONAPI.

Also when doing:
```
$this->setDisableFrontend($config['disablefrontend']);
```
I think you need to add a `isset` check, so it wouldn't break when updating and that value is not present?

**Comment by Raistlfiren** [comment]
@xiaohutai - I was wanting to add the feature because we are only using the Bolt instance for the API to feed child websites. We have no need for the frontend. It would be handy to be able to disable the frontend completely, and only include the backend and jsonapi routes.

I tested having no value in my config for `disableFrontend`, and the value actually returns NULL. 

Is the check necessary? I bet @GawainLynch has his own two cents. ;) lol

**Comment by xiaohutai** [comment]
It's an array, it will give a notice if it's not there:

jsonapi-notice


Maybe:
```
        $disablefrontend = isset($config['disablefrontend']) ? $config['disablefrontend'] : true;
        $this->setDisableFrontend($disablefrontend);
```
what do you think?

**Comment by Raistlfiren** [comment]
I agree @xiaohutai, but I did set the default to false instead of true. Since currently the frontend is always displayed.

**Pull request merged**